### PR TITLE
Added lookups + a few attribute additions/changes

### PR DESCRIPTION
**Pull request by NickShafferNOAA**
Added some lookups for ashop data pull, added a few attributes and changed some existing ones to optional.

These are changes I made awhile ago mostly for the etl stuff and made a pull request for when this was all still in boatnet, that flew under the radar and never got pulled in so I've remade the changes here for this repo.  The old pending PR on boatnet can be deleted after this is resolved.

### CODE REVIEW

**Comment by NickShafferNOAA** [comment]
@BHHorness-NOAA  regarding your questions:

1 - The biostructure array is just to etl the existing dissection data in the wcgop data, without taking into account the current lab results issue, it's likely to change depending on how we end up approaching that.

2 - I think the value for ageLocation is name of the lab, there's only two values found in the data, I am assuming that's what this is.

3 - I think you're right, going forward we will need more than just taxonomy-alias, we could change this to taxonomy-alias or a catch-grouping, or come up with something new specifically for this, calling it "catch-grouping" might be misleading

4 - Comments added to those attributes

5 - I am uncertain. I'm hoping this will be cleared up after talking to the ashop people. I added it to hauls because that's the level it exists at in the present data and in the etl to couch. Melina's shown me earlier today that this attribute will probably need to be changed to encompass more data going forward.

File: bn-models/src/models/_lookups/viability.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { BaseLookup } from '../_base';
```
Comment (wsmith-nwfsc) [inline bn-models/src/models/_lookups/viability.ts]: This is Maturity, but the filename is viability.ts (I think it should be maturity.ts. Be sure to update index.ts as well)

File: bn-models/src/models/ashop/index.ts [hunk 2]
```diff
@@ -3,3 +3,4 @@ export * from './ashop-cruise';
 export * from './ashop-discard-reason';
 export * from './ashop-haul';
 export * from './ashop-trip';
+export * from './ashop-specimen';
```
Comment (wsmith-nwfsc) [inline bn-models/src/models/ashop/index.ts]: The linter will complain about this missing linefeed

File: bn-models/src/models/_lookups/catch-grouping.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ CatchGrouping Type
 
 export interface CatchGrouping extends BaseLookup {
   name: string;
-  code?: number;
+  code?: number | string; // arbitrary code by which this is known by from various programs/systems
```
Comment (wsmith-nwfsc) [inline bn-models/src/models/_lookups/catch-grouping.ts]: Maybe just make this a string to simplify

File: bn-models/src/models/_lookups/index.ts [hunk 4]
```diff
@@ -57,6 +60,7 @@ export * from './vessel-logbook-type';
 export * from './vessel-status';
 export * from './vessel-type';
 export * from './vessel';
+export * from './viability';
```
Comment (wsmith-nwfsc) [inline bn-models/src/models/_lookups/index.ts]: viability -> probably will change to maturity (see below)

File: bn-models/src/models/_lookups/viability.ts (hunk 1 above)
Comment (wsmith-nwfsc) [reply bn-models/src/models/_lookups/viability.ts]: Oh wait, we already have Maturity. 
I think this file needs to be implemented.

**Pull request merged**